src/coreclr/nativeaot/Runtime/unix/UnwindHelpers.cpp

```diff
@@ -104,6 +105,7 @@ struct Registers_REGDISPLAY : REGDISPLAY
         switch (regNum)
         {
         case UNW_REG_IP:
+        case UNW_X86_64_RIP:
```

Comment:
Why do we need to add the UNW_X86_64_RIP? The UNW_REG_IP should be a cross-platform way to represent IP, as far as I know.

Reply:
This is to match the upstream fix - https://github.com/llvm/llvm-project/commit/2f1ee56f3cb8adb5aeadba3f66a8fc39fba1604d

As I understand, UNW_REG_IP is used in code that is not arch-specific, but UNW_X86_64_RIP should still be valid. 

If we have loops like `for (int i = 0; i <= lastReg; ++i) { ` , then it would be operating with UNW_X86_64_RIP and if it is not a valid register, it would cause a failure or be skipped over.

In particular `stepWithDwarf` has a loop like this and was skipping over initialization of RIP in the prologues.

https://github.com/dotnet/runtime/blob/b2dc37ba181a7fa4427e717eab819ba3543d0ae4/src/coreclr/nativeaot/libunwind/src/DwarfInstructions.hpp#L178

Reply:
Ah, ok. Thanks!